tests/embedding/test_app.py

```diff
@@ -86,3 +86,80 @@ def test_warning_message(test_env, rootdir):
 
     # Assert warning message appears for embedded app
     assert warning, "UserWarning should appear in the output of the script"
+
+
+@pytest.mark.embedding
+@pytest.mark.python_env
+def test_private_appdata(test_env, rootdir):
+    """Test embedded instance does not save ShowTriad using a test-scoped Python environment."""
+
+    # Install pymechanical
+    subprocess.check_call(
+        [test_env.python, "-m", "pip", "install", "-e", "."],
+        cwd=rootdir,
+        env=test_env.env,
+    )
+
+    embedded_py = os.path.join(rootdir, "tests", "scripts", "run_embedded_app.py")
+
+    # Set ShowTriad to False
+    set_showtriad = subprocess.Popen(
+        [test_env.python, embedded_py, "True", "Set"],
+        stdout=subprocess.PIPE,
+        env=test_env.env,
+    )
+    set_showtriad.wait()
```

Comment:
imo better to do `subprocess.check_call()` instead of `Popen()` followed by `wait()`

Reply:
I ended up doing check_output for the calls that need the stdout, but I changed the others Popen wait() sections to use check_call instead